server/service/microsoft_mdm_test.go

```diff
@@ -220,3 +221,140 @@ func TestProvisioningDocGeneration(t *testing.T) {
+}
+
+func TestValidSyncMLCmdText(t *testing.T) {
+	testOmaURI := "testuri"
+	testData := "testdata"
+	cmdMsg := NewSyncMLCmdText(mdm_types.CmdReplace, testOmaURI, testData)
+	outXML, err := xml.MarshalIndent(cmdMsg, "", "  ")
+	require.NoError(t, err)
+	require.NotEmpty(t, outXML)
+	payload := string(outXML)
+	err = CheckWrappedSyncMLCmd(fleet.CmdReplace, payload)
+	require.NoError(t, err)
+	require.Contains(t, payload, fmt.Sprintf("<LocURI>%s</LocURI>", testOmaURI))
+	require.Contains(t, payload, fmt.Sprintf("<Data>%s</Data>", testData))
+	require.Contains(t, payload, "<Type xmlns=\"syncml:metinf\">text/plain</Type>")
+	require.Contains(t, payload, "<Format xmlns=\"syncml:metinf\">chr</Format>")
+}
+
+func TestValidSyncMLCmdXml(t *testing.T) {
+	testOmaURI := "testuri"
+	testData := "testdata"
+	cmdMsg := NewSyncMLCmdXml(mdm_types.CmdReplace, testOmaURI, testData)
+	outXML, err := xml.MarshalIndent(cmdMsg, "", "  ")
+	require.NoError(t, err)
+	require.NotEmpty(t, outXML)
+	payload := string(outXML)
+	err = CheckWrappedSyncMLCmd(fleet.CmdReplace, payload)
+	require.NoError(t, err)
+	require.Contains(t, payload, fmt.Sprintf("<LocURI>%s</LocURI>", testOmaURI))
+	require.Contains(t, payload, fmt.Sprintf("<Data>%s</Data>", testData))
+	require.Contains(t, payload, "<Type xmlns=\"syncml:metinf\">text/plain</Type>")
+	require.Contains(t, payload, "<Format xmlns=\"syncml:metinf\">xml</Format>")
+}
+
+func TestValidSyncMLCmdAlert(t *testing.T) {
+	testData := "1234"
+	cmdMsg := NewSyncMLCmdAlert(testData)
+	outXML, err := xml.MarshalIndent(cmdMsg, "", "  ")
+	require.NoError(t, err)
+	require.NotEmpty(t, outXML)
+	payload := string(outXML)
+	err = CheckWrappedSyncMLCmd(fleet.CmdAlert, payload)
+	require.NoError(t, err)
+	require.Contains(t, payload, fmt.Sprintf("<Data>%s</Data>", testData))
+}
+
+func TestValidSyncMLCmd(t *testing.T) {
+	testCmdSource := "testcmdsource"
+	testCmdTarget := "testcmdtarget"
+	testCmdDataType := "testcmddatatype"
+	testCmdDataFormat := "testchr"
+	testCmdDataValue := "testdata"
+	cmdMsg := NewSyncMLCmd(mdm_types.CmdReplace, testCmdSource, testCmdTarget, testCmdDataType, testCmdDataFormat, testCmdDataValue)
+	outXML, err := xml.MarshalIndent(cmdMsg, "", "  ")
+	require.NoError(t, err)
+	require.NotEmpty(t, outXML)
+	payload := string(outXML)
+	err = CheckWrappedSyncMLCmd(fleet.CmdReplace, payload)
+	require.NoError(t, err)
+	require.Contains(t, payload, fmt.Sprintf("<LocURI>%s</LocURI>", testCmdSource))
+	require.Contains(t, payload, fmt.Sprintf("<LocURI>%s</LocURI>", testCmdTarget))
+	require.Contains(t, payload, fmt.Sprintf("<Data>%s</Data>", testCmdDataValue))
+	require.Contains(t, payload, fmt.Sprintf("<Type xmlns=\"syncml:metinf\">%s</Type>", testCmdDataType))
+	require.Contains(t, payload, fmt.Sprintf("<Format xmlns=\"syncml:metinf\">%s</Format>", testCmdDataFormat))
+}
+
+// CheckWrappedSyncMLCmd checks that the payload is wrapped in the given tag.
+func CheckWrappedSyncMLCmd(tag string, data string) error {
```

Comment:
This should probably be unexported as it's just a helper function for this test file.

Reply:
thanks, just fixed this.